face/capture/src/main/java/com/simprints/face/capture/screens/livefeedbackautocapture/LiveFeedbackAutoCaptureFragmentViewModel.kt

```diff
@@ -0,0 +1,261 @@
+import com.simprints.face.infra.basebiosdk.detection.Face
+import com.simprints.face.infra.basebiosdk.detection.FaceDetector
+import com.simprints.face.infra.biosdkresolver.ResolveFaceBioSdkUseCase
+import com.simprints.infra.config.store.models.experimental
+import com.simprints.infra.config.sync.ConfigManager
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.async
+import kotlinx.coroutines.awaitAll
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import java.util.concurrent.atomic.AtomicBoolean
+import javax.inject.Inject
+
+@HiltViewModel
+internal class LiveFeedbackAutoCaptureFragmentViewModel @Inject constructor(
+    private val resolveFaceBioSdk: ResolveFaceBioSdkUseCase,
+    private val configManager: ConfigManager,
+    private val eventReporter: SimpleCaptureEventReporter,
+    private val timeHelper: TimeHelper,
+) : ViewModel() {
+    private var attemptNumber: Int = 1
+    private var samplesToKeep: Int = 1
+    private var qualityThreshold: Float = 0f
+    private var singleQualityFallbackCaptureRequired: Boolean = false
+
+    private val faceTarget = FaceTarget(
+        SymmetricTarget(VALID_YAW_DELTA),
+        SymmetricTarget(VALID_ROLL_DELTA),
+        0.20f..0.5f,
+    )
+    private val fallbackCaptureEventStartTime = timeHelper.now()
+    private var shouldSendFallbackCaptureEvent: AtomicBoolean = AtomicBoolean(true)
+    private var fallbackCapture: FaceDetection? = null
+
+    val userCaptures = mutableListOf<FaceDetection>()
+    var sortedQualifyingCaptures = listOf<FaceDetection>()
+    val currentDetection = MutableLiveData<FaceDetection>()
+    val capturingState = MutableLiveData(CapturingState.NOT_STARTED)
+    private var autoCaptureInitialDelayJob: Job = viewModelScope.launch {
+        delay(AUTO_CAPTURE_VIEWFINDER_RESUME_DELAY_MS)
+    }
+    private lateinit var faceDetector: FaceDetector
+
+    fun startPreparationDelay() {
+        if (capturingState.value != CapturingState.NOT_STARTED) {
+            return // too late - imaging has already started
+        }
+        capturingState.value = CapturingState.NOT_STARTED // reset view
+        autoCaptureInitialDelayJob.cancel()
+        autoCaptureInitialDelayJob = viewModelScope.launch {
+            delay(AUTO_CAPTURE_VIEWFINDER_RESUME_DELAY_MS)
+        }
+    }
+
+    /**
+     * Processes the image
+     *
+     * @param image is the camera frame
+     */
+    fun process(croppedBitmap: Bitmap) {
+        val captureStartTime = timeHelper.now()
+        val potentialFace = faceDetector.analyze(croppedBitmap)
+
+        val faceDetection = getFaceDetectionFromPotentialFace(croppedBitmap, potentialFace)
+        faceDetection.detectionStartTime = captureStartTime
+        faceDetection.detectionEndTime = timeHelper.now()
+
+        if (autoCaptureInitialDelayJob.isCompleted) {
```

Comment:
If delay is not complete, the `faceDetection` is unused. Better to put the if in the beginning of the function.